gpt4all-chat/src/database.cpp

```diff
@@ -1515,8 +1516,22 @@ void Database::handleEmbeddingsGenerated(const QVector<EmbeddingResult> &embeddi
     for (const auto &[key, stat]: std::as_const(stats).asKeyValueRange()) {
         if (!m_collectionMap.contains(key.folder_id)) continue;
         CollectionItem item = guiCollectionItem(key.folder_id);
-        item.currentEmbeddingsToIndex -= stat.nAdded + stat.nSkipped;
-        item.totalEmbeddingsToIndex -= stat.nSkipped;
+        Q_ASSERT(item.currentEmbeddingsToIndex >= stat.nAdded + stat.nSkipped);
+        if (item.currentEmbeddingsToIndex < stat.nAdded + stat.nSkipped) {
+            qWarning() << "Database ERROR: underflow in current embeddings to index statistics";
+            item.currentEmbeddingsToIndex = 0;
+        } else {
+            item.currentEmbeddingsToIndex -= stat.nAdded + stat.nSkipped;
+        }
```

Comment:
Do we not trust this code enough to just do this?
```suggestion
        item.currentEmbeddingsToIndex -= qMax(item.currentEmbeddingsToIndex, stat.nAdded + stat.nSkipped);
```
Or at least, given the number of places where we can warn on underflow, we could make a function to do the subtraction.

Reply:
No, I don't trust it enough that's why I added warnings :) No need to add a function pls. The code works.